Common/Securities/Equity/PatternDayTradingMarginModel.cs

```diff
@@ -0,0 +1,148 @@
+        {
+        }
+
+        /// <summary>
+        /// Gets the total margin required to execute the specified order in units of the account currency including fees
+        /// </summary>
+        /// <remarks>
+        /// If we are in regular market hours, base 4x leverage is used, otherwise leverage is reduced to 2x
+        /// </remarks>
+        /// <param name="security">The security to compute initial margin for</param>
+        /// <param name="order">The order to be executed</param>
+        /// <returns>The total margin in terms of the currency quoted in the order</returns>
+        public override decimal GetInitialMarginRequiredForOrder(Security security, Order order)
+        {
+            var orderFees = security.FeeModel.GetOrderFee(security, order);
+
+            return order.GetValue(security)*InitialMarginRequirement*GetMarginCorrection(security) + orderFees;
+        }
+
+        /// <summary>
+        /// Gets the margin currently alloted to the specified holding
+        /// </summary>
+        /// <param name="security">The security to compute maintenance margin for</param>
+        /// <returns>The maintenance margin required for the </returns>
+        public override decimal GetMaintenanceMargin(Security security)
+        {
+            return base.GetMaintenanceMargin(security)*GetMarginCorrection(security);
+        }
+
+        /// <summary>
+        /// Generates a new order for the specified security taking into account the total margin
+        /// used by the account. Returns null when no margin call is to be issued.
+        /// </summary>
+        /// <param name="security">The security to generate a margin call order for</param>
+        /// <param name="netLiquidationValue">The net liquidation value for the entire account</param>
+        /// <param name="totalMargin">The total margin used by the account in units of base currency</param>
+        /// <returns>An order object representing a liquidation order to be executed to bring the account within margin requirements</returns>
+        public override SubmitOrderRequest GenerateMarginCallOrder(Security security, decimal netLiquidationValue, decimal totalMargin)
+        {
+            // leave a buffer in default implementation
+            const decimal marginBuffer = 0.10m;
+
+            if (totalMargin <= netLiquidationValue * (1 + marginBuffer))
+            {
+                return null;
+            }
+
+            if (!security.Holdings.Invested)
+            {
+                return null;
+            }
+
+            // compute the value we need to liquidate in order to get within margin requirements
+            var delta = totalMargin - netLiquidationValue;
+
+            // compute the number of shares required for the order, rounding up
+            var quantity = (int)(Math.Round(delta / security.Price, MidpointRounding.AwayFromZero) / (MaintenanceMarginRequirement*GetMarginCorrection(security)));
```

Comment:
maybe even the base implementation could be reworked so that this method (on the base) uses GetMaintainenceMargin -- then we probably don't even need to override this method -- thoughts?